### PR TITLE
Set Minimum TLS version to 1.2 for AppChannel's HTTP and gRPC clients

### PR DESCRIPTION
**Pull request by shubham1172**
# Description

Enforce minimum TLS version to 1.2 when Dapr talks to Application over HTTP or gRPC. This is enabled by-default and will break insecure scenarios. Users can still opt-out using a feature flag, however the feature flag will be removed in 2 releases.

## Issue reference

Related issues: https://github.com/dapr/dapr/issues/3565, https://github.com/dapr/dapr/issues/6031

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by shubham1172** [comment]
Question: should be expose this behind a feature flag to allow users to opt-out of TLS 1.2 enforcement? We should remove the feature flag in a few releases then. Or since it's a security issue, are we ok breaking users and forcing them to either be secure (by upgrading to TLS 1.2), or be **intentionally** insecure (by removing --app-ssl flag)?

**Comment by ItalyPaleAle** [comment]
We discussed yesterday in the maintainers call that:

1. We will make TLS >= 1.2 the default. This is a breaking change and we're ok with that.
2. We will add a **temporary** feature flag  (for example `AllowInsecureTLS` - I like that it says "insecure" :) ) for users who want to restore the behavior of allowing TLS 1.0 and 1.1
3. The feature flag will be removed in Dapr 1.13

**Comment by shubham1172** [comment]
Perfect, thank you! I will make these changes @ItalyPaleAle.

File: pkg/channel/http/http_channel.go
Comment (ItalyPaleAle) [inline pkg/channel/http/http_channel.go:71]: Not gonna block the PR due to this, but if you get a chance can you please convert this interface to accept a single struct with options, rather than 8 arguments? It becomes hard to even read what the invocations are doing.

Comment (shubham1172) [reply pkg/channel/http/http_channel.go:71]: Good idea, let me refactor in this PR itself.

**Review by ItalyPaleAle — APPROVED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]
lgtm.
@dapr/maintainers-dapr Any concerns with merging this breaking change for min TLS for 1.11?

**Comment by mukundansundar** [comment]
> lgtm.
> @dapr/maintainers-dapr Any concerns with merging this breaking change for min TLS for 1.11?

based on offline sync there is agreement on this PR being merged.

**Pull request merged**
